ansible/playbooks/roles/repository/library/filter_credentials.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+
+from __future__ import (absolute_import, division, print_function)
+
+from pathlib import Path
+from typing import Callable, Dict, List
+import yaml
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+---
+module: filter_credentials
+
+short_description: Module for filtering sensitive data stored in manifest.yml file
+
+options:
+    src:
+        description: Path to the manifest file that will be filtered
+        required: true
+        type: str
+    dest:
+        description: Path to the newly created, filtered manifest
+        required: false
+        type: str
+    in_place:
+        description: When set to True will modify existing manifest passed as `src`
+        required: false
+        type: bool
+"""
+
+EXAMPLES = r"""
+# Pass in a manifest file without modifying the original file
+- name: Filter out manifest file and set result to stdout
+  filter_credentials:
+    src: /some/where/manifest.yml
+
+# Pass in a manifest file and modify it in place
+- name: Filter out manifest file and replace the original file
+  filter_credentials:
+    src: /some/where/manifest.yml
+    in_place: true
+
+# Pass in a manifest file and save it to `dest` location
+- name: Filter out manifest file and save it as a new file
+  filter_credentials:
+    src: /some/where/manifest.yml
+    dest: /some/other/place/manifest.yml
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def _filter_common(docs: List[Dict]):
+    # remove admin user info from epiphany-cluster doc:
+    del next(filter(lambda doc: doc['kind'] == 'epiphany-cluster', docs))['specification']['admin_user']
+
+
+def _filter_aws(docs: List[Dict]):
+    _filter_common(docs)
+
+    # filter epiphany-cluster doc
+    epiphany_cluster = next(filter(lambda doc: doc['kind'] == 'epiphany-cluster', docs))
+    del epiphany_cluster['specification']['cloud']['credentials']
+
+
+def _filter_azure(docs: List[Dict]):
+    _filter_common(docs)
+
+    # filter epiphany-cluster doc
+    epiphany_cluster = next(filter(lambda doc: doc['kind'] == 'epiphany-cluster', docs))
+    del epiphany_cluster['specification']['cloud']['subscription_name']
+
+
+def _get_filtered_manifest(manifest_path: Path) -> str:
+    """
+    Load the manifest file and remove any sensitive data.
+
+    :param manifest_path: manifest file which will be loaded
+    :returns: filtered manifset
+    """
+    docs = yaml.safe_load_all(manifest_path.open())
+    filtered_docs = [doc for doc in docs if doc['kind'] in ['epiphany-cluster',
+                                                            'configuration/feature-mappings',
+                                                            'configuration/image-registry']]
+
+    FILTER_DATA: Dict[str, Callable] = {
+        'any': _filter_common,
+        'azure': _filter_azure,
+        'aws': _filter_aws
+    }
+
+    FILTER_DATA[filtered_docs[0]['provider']](filtered_docs)
```

Comment:
@seriva can input manifest contain multiple providers or we always have just one?

Reply:
We support only single provider in a manifest.